scripts/post_migration_fixes/pmf_reports_20220308_in1197/annual_report_logs.sql

```diff
@@ -0,0 +1,84 @@
+CREATE SCHEMA IF NOT EXISTS pmf_reports_20220308_in1197;
+
+-- DROP TABLE IF EXISTS pmf_reports_20220308_in1197.affected_cases;
+-- DROP TABLE IF EXISTS pmf_reports_20220308_in1197.annual_report_logs_deletes;
+-- DROP TABLE IF EXISTS pmf_reports_20220308_in1197.scheduled_events_inserts;
+
+-- Capture affected cases that have multiple pending reports
+SELECT count(*), ar.client_id
+INTO pmf_reports_20220308_in1197.affected_cases
+FROM annual_report_logs ar
+INNER JOIN persons p ON p.id = ar.client_id
+WHERE p.clientsource = 'CASRECMIGRATION'
+AND ar.status = 'PENDING'
+GROUP BY ar.client_id
+HAVING count(*) > 1;
+
+-- The last Pending report should be deleted where there are multiple Pending reports
```

Comment:
Would this delete all PENDING annual_report_logs for any client with 2 or more rows in that table, or does it leave one of the rows behind? It looks like it will just remove all annual_report_logs but maybe I've missed something.

Reply:
SELECT DISTINCT ON is the bit that makes it just pull back a single record. It's similar to doing a OVER and partition by and order by and bringing back row 1.

Reply:
Is that the correct logic here? It probably is but this is what is being deleted so you're deleting 1 record for each clientid which has the latest reportingperiodenddate date. 

Just double checking that:
a) there is never more than one to delete as this won't do that
b) it's definitely the latest one that we want to delete and not the earliest

I'm sure that is all right, I'm just double checking :)